src/sisl/io/sile.py

```diff
@@ -1574,13 +1614,45 @@ class SileBin(BaseSile):
     """
 
     def __init__(self, filename, mode="r", *args, **kwargs):
-        self._file = Path(filename)
         # Open mode
         self._mode = mode.replace("b", "") + "b"
+        # Get file
+        self._file = self._sanitize_filename(filename)
+
+        self._is_inside_zip = isinstance(self.file, ZipPath)
+        self._temp_file_written = False
+        self._raw_file = self._file
 
         # Must call setup-methods
         self._base_setup(*args, **kwargs)
 
+    def __getattribute__(self, name):
+        # If the file is inside a zip fortran can't read directly from it.
+        # We therefore store the contents in a temporary file and set this
+        # as the file to be read from.
+        if (
+            name.startswith("read_")
```

Comment:
what about write? Should that even be allowed?

Reply:
Yes, I think so. It is a bit more scary because if we don't manage the write correctly it can end up in a corrupted zip file, that's why I didn't want to implement it fast, I need to put a bit of thought into it.

Reply:
Otherwise, doing an `NotImplementedError` is sufficiently at the moment, it doesn't seem like it is something that's very useful ATM.

Reply:
Yes but on the other hand if I do it now it will be faster than if I come back at it 2 years later :sweat_smile: